manifests/oracle.pp

```diff
@@ -0,0 +1,215 @@
+    '6' : {
+      $releaseMajor = '6u45'
+      $releaseMinor = 'b06'
+      $installPath = "${javaSE}1.6.0_45"
+    }
+    '7' : {
+      $releaseMajor = '7u80'
+      $releaseMinor = 'b15'
+      $installPath = "${javaSE}1.7.0_80"
+    }
+    '8' : {
+      $releaseMajor = '8u51'
+      $releaseMinor = 'b16'
+      $installPath = "${javaSE}1.8.0_51"
+    }
+    default : {
+      $releaseMajor = '8u51'
+      $releaseMinor = 'b16'
+      $installPath = "${javaSE}1.8.0_51"
+    }
+  }
+
+  # determine package type (exe/tar/rpm), destination directory based on OS
+  case downcase($::kernel) {
+    'linux' : {
+      case downcase($::osfamily) {
+        'redhat', 'centos' : {
+          if $version == '6' {
+            $packageType = 'rpmbin'
+          } else {
+            $packageType = 'rpm'
+          }
+        }
+        default : {
+          fail ("OS ${::osfamily} unsupported platform.") }
+      }
+
+      $os = 'linux'
+      $destinationDir = '/tmp/'
+      $createsPath = "/usr/java/${installPath}"
+    }
+    default : {
+      fail ( "Kernel ${::kernel} unsupported platform." ) }
+  }
+
+  # determine java architecture type
+  case $::architecture {
+    'i386' : { $arch = 'i586' }
+    'x86_64' : { $arch = 'x64' }
+    default : {
+      fail ("Architecture ${::architecture} unsupported platform.")
+    }
+  }
+
+  # following are based on this example:
+  # http://download.oracle.com/otn/java/jdk/7u80-b15/jre-7u80-linux-i586.rpm
+  #
+  # JaveSE 6 distributed in .bin format
+  # http://download.oracle.com/otn/java/jdk/6u45-b06/jdk-6u45-linux-i586-rpm.bin
+  # http://download.oracle.com/otn/java/jdk/6u45-b06/jdk-6u45-linux-i586.bin
+  # package name to download from Oracle's website
+  case $packageType {
+    'bin' : {
+      $packageName = "${javaSE}-${releaseMajor}-${os}-${arch}.bin"
+    }
+    'rpmbin' : {
+      $packageName = "${javaSE}-${releaseMajor}-${os}-${arch}-rpm.bin"
+    }
+    'rpm' : {
+      $packageName = "${javaSE}-${releaseMajor}-${os}-${arch}.rpm"
+    }
+    default : {
+      $packageName = "${javaSE}-${releaseMajor}-${os}-${arch}.rpm"
+    }
+  }
+
+  # full path to the installer
+  $destination = "${destinationDir}${packageName}"
+  notice ("Destination is ${destination}")
+
+  case $packageType {
+    'bin' : {
+      $installCommand = "sh ${destination}"
+    }
+    'rpmbin' : {
+      $installCommand = "sh ${destination} -x; rpm --force -iv sun*.rpm; rpm --force -iv ${javaSE}*.rpm"
+    }
+    'rpm' : {
+      $installCommand = "rpm --force -iv ${destination}"
+    }
+    default : {
+      $installCommand = "rpm -iv ${destination}"
+    }
+  }
+
+  # full URL to download
+  $url = "${baseUrl}${releaseMajor}-${releaseMinor}/${packageName}"
+
+  case $ensure {
+    'present' : {
+      archive { $destination :
+        ensure  => present,
+        source  => $url,
+        cleanup => false,
+        cookie  => 'gpw_e24=http%3A%2F%2Fwww.oracle.com%2F; oraclelicense=accept-securebackup-cookie',
```

Comment:
This is circumventing the license agreement, yeah? So... not something we should advocate.